<launch>
  <arg name="show_windows" default="false"/>
  <node pkg="vision" type="cascade_classifier_node" name="detector_node" ns="aero/lower_stereo" output="screen">
    <param name="show_windows" value="$(arg show_windows)"/>
    <param name="robot_frame" value="aero/base_footprint"/>

    <rosparam param="training" subst_value="True">[
{id: 0, label: "hook",
cascade_path: "$(find vision)/training/hook_cascade.xml",
object_radius: 0.03,
scale_factor: 1.01, min_neighbors: 10,
min_size: {width: 50.0, height: 50.0}, max_size: {width: 150.0, height: 150.0},
disparity_size: {width: 20.0, height: 40.0}, disparity_y_offset: 20,
max_disparity_for_detection: 98.0},
    ]</rosparam>
  </node>

  <node pkg="robot_localization" type="ekf_localization_node" name="detection_ekf" ns="aero/lower_stereo" output="screen">
    <param name="pose0" value="object_detection"/>
    <rosparam param="pose0_config">[true, true, true, true, true, true]</rosparam>
    <rosparam param="pose0_differential">[false, false, false, false, false, false]</rosparam>

    <param name="twist0" value="object_detection_zero_twist"/>
    <rosparam param="twist0_config">[true, true, true, true, true, true]</rosparam>
    <rosparam param="twist0_differential">[false, false, false, false, false, false]</rosparam>

    <param name="odom_frame" value="aero/odom"/>
    <param name="base_link_frame" value="aero/sample_detection_ekf"/>

    <remap from="odometry/filtered" to="object_detection/filtered_odom"/>
  </node>


  <node pkg="vision" type="zero_twist_publisher" name="zero_twist_publisher" ns="aero/lower_stereo" output="screen">
    <remap from="zero_twist" to="object_detection_zero_twist"/>
  </node>


  <node pkg="vision" type="detection_odom_filter_node" name="odom_filter" ns="aero/lower_stereo/object_detection" output="screen">
    <param name="covar_limit" value="0.2"/>
    <param name="frame_id" value="aero/odom"/>
    <param name="child_frame_id" value="aero/sample_detection"/>
    <param name="invert_tf" value="false"/>
    <remap from="detection_odom" to="filtered_odom"/>
    <remap from="detection" to="filtered"/>
  </node>
</launch>
